### PR TITLE
Add common form step styles, bring protocol elements into steps

### PR DESCRIPTION
**Pull request by hannajones**
This PR provides a way to pull Protocol styles/other common styles into all of our form steps via `BaseFormStep`. It also adds some styling/polish to the existing form steps. Here's how things are looking now:

Screenshot 2023-05-17 at 11 02 13 AM

Screenshot 2023-05-17 at 11 02 00 AM

### CODE REVIEW

**Comment by hannajones** [comment]
@mikeconley I just realized protocol doesn't seem to provide the `:invalid` form styles from the mockups 🤔 Might take a stab at that here, in which case more changes incoming

File: kitsune/sumo/static/sumo/scss/form-wizard-configuration-step.styles.scss [hunk 1]
```diff
@@ -54,4 +63,16 @@
 
 #next {
   align-self: end;
+  margin-block-end: 16px;
+  margin-block-start: 8px;
```
Comment (mikeconley) [inline kitsune/sumo/static/sumo/scss/form-wizard-configuration-step.styles.scss:67]:
```suggestion
  margin-block: 8px 16px;
```

**Review by mikeconley — APPROVED** [review]
Brilliant as usual - just one small suggestion before landing. Thanks @hannajones!

**Comment by hannajones** [comment]
Small update - I filed a separate ticket for the custom validation message, since it's a decent chunk of work and should probably be reviewed separately: https://github.com/mozilla/sumo/issues/1305

**Comment by hannajones** [comment]
> > @mikeconley I just realized protocol doesn't seem to provide the `:invalid` form styles from the mockups 🤔 Might take a stab at that here, in which case more changes incoming
> 
> Ah, just saw this part of the pull request. Okay, I'll re-review once you add these.

I'm actually going to tackle those changes in a separate patch - there's a decent chunk of JS work involved so it feels a little distinct from these other style changes

**Pull request merged**